File: 136-single-number/single-number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int n=nums.size();
        int mx=nums[0];
        int mi=nums[0];
        for(int i=0;i<n;i++){
            mx=max(mx,nums[i]);
            mi=min(mi,nums[i]);
        }
        vector<int> hash(mx-mi+1);
        for(int i=0;i<n;i++){
            hash[nums[i]-mi]++;
        }
        for(int i=0;i<n;i++){
            if(hash[nums[i]-mi]==1){
                return nums[i];
            }
        }
        return -1;
    }
};

File: 136-single-number/single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        int n=nums.length;
        int max=nums[0];
        int min=nums[0];

        for(int i=0;i<n;i++){
          max=Math.max(max,nums[i]);
          min=Math.min(min,nums[i]);
        }
        int[] hash=new int[max-min+1];
        for(int i=0;i<n;i++){
          hash[nums[i]-min]++;
        }
        for(int i=0;i<n;i++){
          if(hash[nums[i]-min]==1){
            return nums[i];
          }
        }
        return -1;
    }
}